python_modules/dagster/dagster/core/execution.py

```diff
@@ -620,7 +620,7 @@ def execute_reentrant_pipeline(pipeline, typed_environment, throw_on_error, reen
 def get_subset_pipeline(pipeline, solid_subset):
     check.inst_param(pipeline, 'pipeline', PipelineDefinition)
     check.opt_list_param(solid_subset, 'solid_subset', of_type=str)
-    return pipeline if solid_subset is None else build_sub_pipeline(pipeline, solid_subset)
+    return pipeline if not solid_subset else build_sub_pipeline(pipeline, solid_subset)
```

Comment:
Yep that is correct - I can change this back, I think it hard-crashes if you create a zero-solid pipeline so I figured it might as well assume the empty case meant `all`.

Reply:
Ahh so I remember now... I changed this because I didn't want to implement `string[] | null` on the JS side 😅  I changed this in the Python and updated the JS side to match (which I think is best for consistency, even if the `string[] | null` is annoying in a few placecs. Should be good now.